src/azure/com/microsoftopentechnologies/intellij/wizards/WizardCacheManager.java

```diff
@@ -467,22 +467,25 @@ public static void cachePublishData(File publishSettingsFile, PublishData publis
 					future.get(OPERATIONS_TIMEOUT, TimeUnit.SECONDS);
 				}
 
-				for (Subscription sub : publishData.getPublishProfile().getSubscriptions()) {
-					/*
-					 * Get collection of storage services in each subscription.
-					 */
-					StorageServices services = publishData.getStoragesPerSubscription().get(sub.getId());
-					if (services != null) {
-						for (StorageService strgService : services) {
-							List<URI> endpoints = strgService.getStorageAccountProperties().getEndpoints();
-                            for (int i = 0; i < endpoints.size(); i++) {
-                                String uri = endpoints.get(i).toString();
-                                if (uri.startsWith("https://")) {
-                                    endpoints.set(i, URI.create(uri.replaceFirst("https://", "http://")));
-                                }
-                            }
+				try {
+					String chinaMngmntUrl = PreferenceSetUtil.getManagementURL("windowsazure.cn (China)", AzurePlugin.prefFilePath);
+					chinaMngmntUrl = chinaMngmntUrl.substring(0, chinaMngmntUrl.lastIndexOf("/"));
+					if (url.equals(chinaMngmntUrl)) {
+						for (Subscription sub : publishData.getPublishProfile().getSubscriptions()) {
+							StorageServices services = publishData.getStoragesPerSubscription().get(sub.getId());
+							for (StorageService strgService : services) {
+								List<URI> endpoints = strgService.getStorageAccountProperties().getEndpoints();
+								for (int i = 0; i < endpoints.size(); i++) {
+									String uri = endpoints.get(i).toString();
+									if (uri.startsWith("https://")) {
+										endpoints.set(i, URI.create(uri.replaceFirst("https://", "http://")));
+									}
+								}
+							}
 						}
 					}
+				} catch (Exception e) {
+					// Ignore error
```

Comment:
If some exception occurs while retrieving management URL or if management URL is empty/NULL then "Management URL does not exist." exception will be thrown by PreferenceSetUtil.getManagementURL() function call.
Just to be on safer side we can log the exception (otherwise I don’t see any issues even though we do not log it).